--- ../ArduinoEthernet/EthernetDNS/EthernetDNS.cpp	2010-01-21 12:54:52.000000000 +0900
+++ EthernetDNS.cpp	2010-03-14 03:55:16.000000000 +0900
@@ -367,6 +367,11 @@
                    read_data(this->_socket, (vuint8*)(ptr+offset), (vuint8*)ipAddr, 4);
                    statusCode = DNSSuccess;
                    break;
+                /*** Support type CNAME from here by H.Yamada ***/
+                // if "type CNAME" match, skip next Answer.
+                } else if (5 == buf[1]) {
+                   offset += buf[5];
+                /*** Support type CNAME to here by H.Yamada ***/
                 }
              } else
                 offset += 4; // eat the query type and class fields of a query
